src/ellipsoid.jl

```diff
@@ -25,7 +25,31 @@ Base.hash(b::Ellipsoid, h::UInt) = hash(b.c, hash(b.ri2, hash(b.p, hash(b.data,
 
 Base.in(x::SVector{N}, b::Ellipsoid{N}) where {N} = sum((b.p * (x - b.c)).^2 .* b.ri2) ≤ 1.0
 
-normal(x::SVector{N}, b::Ellipsoid{N}) where {N} = normalize(Ac_mul_B(b.p, b.ri2 .* (b.p * (x - b.c))))
+function surfpt_nearby(x::SVector{N}, b::Ellipsoid{N}) where {N}
+    # For a given point x and equation of ellipsoid f(x) = 1, find t such that x₀ = x + t*∇f(x)
+    # is on the ellipsoid.  Eventually this reduces to a quadratic equation for t.  The
+    # following is evaluation of the quadratic formula in a numerically stable way.
+    px = b.p * (x - b.c)  # in ellipsoid's coordinates
+    px2 = px.^2
+
+    px²r⁻² = px2 ⋅ b.ri2
+    px²r⁻⁴ = px2 ⋅ (b.ri2.^2)
+    px²r⁻⁶ = px2 ⋅ (b.ri2.^3)
+
+    q24 = (px²r⁻² - 1) / px²r⁻⁴
+    q64 = px²r⁻⁶ / px²r⁻⁴
```

Comment:
Yes, indeed...  Thanks for pointing this out!  Just filled the hole and added unit tests for `x = (center)` for all `Shape`s.